fineract-core/src/main/java/org/apache/fineract/infrastructure/core/filters/GeolocationHeaderFilter.java

```
@@ -0,0 +1,95 @@
+
+    private static final String[] IP_HEADER_CANDIDATES = { "X-Forwarded-For", "Proxy-Client-IP", "WL-Proxy-Client-IP",
+            "HTTP_X_FORWARDED_FOR", "HTTP_X_FORWARDED", "HTTP_X_CLUSTER_CLIENT_IP", "HTTP_CLIENT_IP", "HTTP_FORWARDED_FOR",
+            "HTTP_FORWARDED", "HTTP_VIA", "REMOTE_ADDR" };
+
+    public static String getClientIpAddress(HttpServletRequest request) {
+        for (String header : IP_HEADER_CANDIDATES) {
+            String ip = request.getHeader(header);
+            if (ip != null && ip.length() != 0 && !"unknown".equalsIgnoreCase(ip)) {
+                log.debug("CALLER IP : {}", ip);
+                return ip;
+            }
+        }
+        log.debug("getRemoteAddr method : {}", request.getRemoteAddr());
+        return request.getRemoteAddr();
+    }
+
+    @Override
+    protected void doFilterInternal(final HttpServletRequest request, final HttpServletResponse response, final FilterChain filterChain)
+            throws IOException, ServletException {
+        FineractProperties.FineractGeolocationProperties geolocationProperties = fineractProperties.getGeolocation();
+        if (geolocationProperties.isEnabled()) {
+            handleClientIp(request, response, filterChain, geolocationProperties);
+        } else {
+            filterChain.doFilter(request, response);
+        }
+
+    }
+
+    private void handleClientIp(HttpServletRequest request, HttpServletResponse response, FilterChain filterChain,
+            FineractProperties.FineractGeolocationProperties geolocationProperties) throws IOException, ServletException {
+        try {
+            String clientIpAddress = getClientIpAddress(request);
+            if (StringUtils.isNotBlank(clientIpAddress)) {
+                log.debug("Found Client IP in header : {}", clientIpAddress);
+                ClientIpHolder.setClientIp(clientIpAddress);
+            }
+            filterChain.doFilter(request, response);
+        } catch (Exception e) {
```

This try - catch is way too broad! You should move the `filterChain.doFilter(request, response)` call outside as this will catch and hide any exceptions that might come from any other filters! 

Do we need the catch for anything? I think it should be removed all together!

---

Thanks for the heads-up! I removed the broad catch—now filterChain.doFilter() is only inside a try {…} finally to guarantee ClientIpHolder.clear(), and any exceptions propagate normally.